bug: Non-Deterministic Test JoinGroupAuthzIT.shouldEnforceAccessToTopics

A test failed non-deterministically in GitHub Actions:

## failed test

`io.kroxylicious.it.filter.authorization.JoinGroupAuthzIT.shouldEnforceAccessToTopics(VersionSpecificVerification)[1]`

https://github.com/kroxylicious/kroxylicious/actions/runs/24478633738/job/71538061248?pr=3724

## stack trace

```
[ERROR] Tests run: 20, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 21.21 s <<< FAILURE! -- in io.kroxylicious.it.filter.authorization.JoinGroupAuthzIT
[ERROR] io.kroxylicious.it.filter.authorization.JoinGroupAuthzIT.shouldEnforceAccessToTopics(VersionSpecificVerification)[1] -- Time elapsed: 1.015 s <<< FAILURE!
org.opentest4j.AssertionFailedError: 
[Expect proxied response to be the same as the reference response with equivalent AuthZ] JSON documents are different:
Different value found in node "alice", expected: <"{
  "errorCode" : 0,
  "generationId" : 1,
  "protocolName" : "range",
  "leader" : "CLOBBERED",
  "memberId" : "CLOBBERED",
  "members" : [ {
    "memberId" : "CLOBBERED",
    "metadata" : ""
  } ]
}"> but was: <"{
  "errorCode" : 16,
  "generationId" : -1,
  "protocolName" : "",
  "leader" : "CLOBBERED",
  "memberId" : "CLOBBERED",
  "members" : [ ]
}">
```

The test expected error code 0 (success) but received error code 16 (GROUP_COORDINATOR_NOT_AVAILABLE), suggesting a potential timing or race condition.

--- a/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/JoinGroupAuthzIT.java
+++ b/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/JoinGroupAuthzIT.java
@@ -21,8 +21,6 @@ import org.apache.kafka.common.acl.AccessControlEntry;
 import org.apache.kafka.common.acl.AclBinding;
 import org.apache.kafka.common.acl.AclOperation;
 import org.apache.kafka.common.acl.AclPermissionType;
-import org.apache.kafka.common.message.ConsumerGroupHeartbeatRequestData;
-import org.apache.kafka.common.message.ConsumerGroupHeartbeatResponseData;
 import org.apache.kafka.common.message.JoinGroupRequestData;
 import org.apache.kafka.common.message.JoinGroupResponseData;
 import org.apache.kafka.common.protocol.ApiKeys;
@@ -157,6 +155,11 @@ public class JoinGroupAuthzIT extends AuthzIT {
         }
 
         @Override
+        public boolean needsRetry(JoinGroupResponseData response) {
+            return Errors.forCode(response.errorCode()) == Errors.NOT_COORDINATOR;
+        }
+
+        @Override
         public void prepareCluster(BaseClusterFixture cluster) {
             KafkaDriver driver = new KafkaDriver(cluster, cluster.authenticatedClient(AuthzIT.SUPER, SUPER_PASSWORD), AuthzIT.SUPER);
             driver.findCoordinator(FindCoordinatorRequest.CoordinatorType.GROUP, group);
@@ -211,7 +214,7 @@ public class JoinGroupAuthzIT extends AuthzIT {
 
     @ParameterizedTest
     @MethodSource
-    void shouldEnforceAccessToTopics(VersionSpecificVerification<ConsumerGroupHeartbeatRequestData, ConsumerGroupHeartbeatResponseData> test) {
+    void shouldEnforceAccessToTopics(VersionSpecificVerification<JoinGroupRequestData, JoinGroupResponseData> test) {
         try (var referenceCluster = new ReferenceCluster(kafkaClusterWithAuthz, this.topicIdsInUnproxiedCluster);
                 var proxiedCluster = new ProxiedCluster(kafkaClusterNoAuthz, this.topicIdsInProxiedCluster, rulesFile)) {
             test.verifyBehaviour(referenceCluster, proxiedCluster);

